File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        int n=s.size();
        stack<char>st;
        string ans="";
        for(int i=0;i<n;i++){
           if(s[i]=='('){
                if(st.empty()){
                    st.push(s[i]);
                }else{
                    ans+=s[i];
                    st.push(s[i]);
                }
           }else{
                st.pop();
                if(!st.empty()){
                    ans+=s[i];
                }
           }
        }
        return ans;
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int n=s.length();
        String ans="";
        int cnt=0;
        for(int i=0;i<n;i++){
           if(s.charAt(i)==')'){
            cnt--;
           }
           if(cnt!=0){
            ans+=s.charAt(i);
           }
           if(s.charAt(i)=='('){
            cnt++;
           }
        }

        return ans;
    }
}